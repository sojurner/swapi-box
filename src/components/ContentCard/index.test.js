import React from "react";

import { ContentCard } from "./index";

describe("ContentCard", () => {
  let wrapper;
  let mockPersonData;
  let mockPlanetData;
  let mockVehicleData;
  let mockToggleFavorites;

  beforeEach(() => {
    mockPersonData = {
      name: "Luke Skywalker",
      species: "Human",
      homeworld: "Tatooine",
      population: "200000",
      favorite: false
    };

    mockPlanetData = {
      climate: "temperate",
      favorite: false,
      name: "Alderaan",
      populaton: "2000000000",
      residents: ["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"],
      terrain: "grasslands, mountains"
    };

    mockVehicleData = {
      name: "Sand Crawler",
      model: "Digger Crawler",
      class: "wheeled",
      numberof_passengers: "30",
      favorite: false
    };

    mockToggleFavorites = jest.fn();

    wrapper = shallow(
      <ContentCard
        key={0}
        card={mockPersonData}
        toggleFavorites={mockToggleFavorites}
      />
    );
  });

  it("should match snapshot with people data object passed", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with planet data object passed", () => {
    wrapper = shallow(
    <ContentCard 
      key={0} 
        card={mockPlanetData}
        toggleFavorites={mockToggleFavorites}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with vehicle data object passed", () => {
    wrapper = shallow(
      <ContentCard
        key={0}
        card={mockVehicleData}
        toggleFavorites={mockToggleFavorites}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with Person favorite status true", () => {
    mockPersonData = {
      name: "Luke Skywalker",
      species: "Human",
      homeworld: "Tatooine",
      population: "200000",
      favorite: true
    };
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with Planet favorite status true", () => {
    mockPlanetData = {
      climate: "temperate",
      favorite: true,
      name: "Alderaan",
      populaton: "2000000000",
      residents: ["Leia Organa", "Bail Prestor Organa", "Raymus Antilles"],
      terrain: "grasslands, mountains"
    };

    wrapper = shallow(
      <ContentCard
        key={0}
        card={mockPlanetData}
        toggleFavorites={mockToggleFavorites}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with favorite status true", () => {
    mockData = {name: "Luke Skywalker", species: "Human", homeworld: "Tatooine", population: "200000", favorite: true}
    expect(wrapper).toMatchSnapshot();
  });
});