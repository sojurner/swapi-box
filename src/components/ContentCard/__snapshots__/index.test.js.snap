// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard should match snapshot with Person favorite status true 1`] = `
<article
  className="content_card"
>
  <h1
    key="name0"
  >
    Luke Skywalker
  </h1>
  <p
    key="species1"
  >
    species
    : 
    Human
  </p>
  <p
    key="homeworld2"
  >
    homeworld
    : 
    Tatooine
  </p>
  <p
    key="population3"
  >
    population
    : 
    200000
  </p>
  <button
    className="button"
    onClick={[Function]}
  />
</article>
`;

exports[`ContentCard should match snapshot with Person favorite status true 2`] = `
<article
  className="content_card favorite"
>
  <p
    key="climate0"
  >
    climate
    : 
    temperate
  </p>
  <h1
    key="name2"
  >
    Alderaan
  </h1>
  <p
    key="populaton3"
  >
    populaton
    : 
    2000000000
  </p>
  <p
    key="residents4"
  >
    residents
    : 
    Leia Organa
    Bail Prestor Organa
    Raymus Antilles
  </p>
  <p
    key="terrain5"
  >
    terrain
    : 
    grasslands, mountains
  </p>
  <button
    className="button"
    onClick={[Function]}
  />
</article>
`;

exports[`ContentCard should match snapshot with Person favorite status true 3`] = `
<article
  className="content_card favorite"
>
  <h1
    key="name0"
  >
    Sand Crawler
  </h1>
  <p
    key="model1"
  >
    model
    : 
    Digger Crawler
  </p>
  <p
    key="class2"
  >
    class
    : 
    wheeled
  </p>
  <p
    key="number of passengers3"
  >
    number of passengers
    : 
    30
  </p>
  <button
    className="button"
    onClick={[Function]}
  />
</article>
`;

exports[`ContentCard should match snapshot with people data object passed 1`] = `
<article
  className="content_card"
>
  <h1
    key="name0"
  >
    Luke Skywalker
  </h1>
  <p
    key="species1"
  >
    species
    : 
    Human
  </p>
  <p
    key="homeworld2"
  >
    homeworld
    : 
    Tatooine
  </p>
  <p
    key="population3"
  >
    population
    : 
    200000
  </p>
  <button
    className="button"
    onClick={[Function]}
  />
</article>
`;

exports[`ContentCard should match snapshot with planet data object passed 1`] = `
<article
  className="content_card"
>
  <p
    key="climate0"
  >
    climate
    : 
    temperate
  </p>
  <h1
    key="name2"
  >
    Alderaan
  </h1>
  <p
    key="populaton3"
  >
    populaton
    : 
    2000000000
  </p>
  <p
    key="residents4"
  >
    residents
    : 
    Leia Organa
    Bail Prestor Organa
    Raymus Antilles
  </p>
  <p
    key="terrain5"
  >
    terrain
    : 
    grasslands, mountains
  </p>
  <button
    className="button"
    onClick={[Function]}
  />
</article>
`;

exports[`ContentCard should match snapshot with vehicle data object passed 1`] = `
<article
  className="content_card"
>
  <h1
    key="name0"
  >
    Sand Crawler
  </h1>
  <p
    key="model1"
  >
    model
    : 
    Digger Crawler
  </p>
  <p
    key="class2"
  >
    class
    : 
    wheeled
  </p>
  <p
    key="number of passengers3"
  >
    number of passengers
    : 
    30
  </p>
  <button
    className="button"
    onClick={[Function]}
  />
</article>
`;
